### PR TITLE
New SBClient Stuff

### PR DESCRIPTION
**Pull request by mglagola**
### Not Ready

### CODE REVIEW

**Comment by joshholat** [comment]
This looks pretty good to me :+1:

File: Projects/Library/CocoaBloc/API/SBClient+Account.m [hunk 1]
```diff
@@ -123,4 +124,21 @@ - (RACSignal*)unfollowAccountWithIdentifier:(NSNumber*)identifier {
...
 
+- (RACSignal *)getAuthenticatedUserAccountsWithParameters:(NSDictionary*)parameters {
+    
+    RACSignal *requestSignal = [[self rac_GET:@"accounts" parameters:[self requestParametersWithParameters:parameters]]
+                                 cb_deserializeArrayWithClient:self keyPath:@"data"];
+
+    //if we are getting admin accounts, save them
+    if ([[parameters objectForKey:@"admin"] boolValue]) {
+        @weakify(self);
+        [requestSignal doNext:^(NSArray *accounts) {
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/API/SBClient+Account.m]: This `doNext:` will never be called, as you're not assigning the result of the operator to anything. You're returning the `requestSignal` without this side effect, below.

Comment (mglagola) [inline Projects/Library/CocoaBloc/API/SBClient+Account.m]: @JRHeaton true - ill fix this

File: Projects/Library/CocoaBloc/API/SBClient+Status.m [hunk 2]
```diff
@@ -7,7 +7,36 @@
...
+
+- (RACSignal *)postStatus:(NSString *)status
+  toAccountWithIdentifier:(NSNumber*)accountIdentifier
+               fanContent:(BOOL)fanContent
+                 latitude:(NSNumber*)latitude
+                longitude:(NSNumber*)longitude {
+    NSParameterAssert(status);
+    NSParameterAssert(accountIdentifier);
+    
+    NSMutableDictionary *params = @{@"text" : status,
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/API/SBClient+Status.m]: Use an `NSMutableArray` initializer here rather than copying a literal, which is a runtime cost.

**Comment by JRHeaton** [comment]
For stylistic consistency, let's keep pointers in the format of `Type *name` with the asterisk on the variable name. There are a couple of instances like `NSString*` throughout this PR.

File: Projects/Library/CocoaBloc/API/SBClient+Content.m [hunk 3]
```diff
@@ -45,4 +52,34 @@ - (RACSignal *)deleteContent:(SBContent *)content {
                 cb_deserializeWithClient:self keyPath:@"data"];
 }
 
+- (RACSignal *)getContentWithIdentifier:(NSNumber *)identifier type:(NSString*)type forAccountWithIdentifier:(NSNumber *)accountIdentifier {
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/API/SBClient+Content.m]: Pointer style

File: Projects/Library/CocoaBloc/API/SBClient+Push.h [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Mark Glagola on 1/26/15.
+//  Copyright (c) 2015 StageBloc. All rights reserved.
+//
+
+#import "SBClient.h"
+
+@interface SBClient (Push)
+
+- (RACSignal *)setPushTokenForAuthenticatedUser:(NSString*)token;
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/API/SBClient+Push.h]: Pointer style

File: Projects/Library/CocoaBloc/API/SBClient+Status.h [hunk 5]
```diff
@@ -7,8 +7,22 @@
...
+/*!
+ Post status to account. Convenience method for posting statuses
+ */
+- (RACSignal *)postStatus:(NSString *)status toAccountWithIdentifier:(NSNumber*)accountIdentifier fanContent:(BOOL)fanContent;
+
+/*!
+ Post status to account.
+ */
+- (RACSignal *)postStatus:(NSString *)status
+  toAccountWithIdentifier:(NSNumber*)accountIdentifier
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/API/SBClient+Status.h]: Pointer style

File: Projects/Library/CocoaBloc/API/SBClient+Account.m [hunk 6]
```diff
@@ -123,4 +124,21 @@ - (RACSignal*)unfollowAccountWithIdentifier:(NSNumber*)identifier {
                 setNameWithFormat:@"Unfollow account %d", identifier.intValue];
 }
 
+- (RACSignal *)getAuthenticatedUserAccountsWithParameters:(NSDictionary*)parameters {
```
Comment (JRHeaton) [inline Projects/Library/CocoaBloc/API/SBClient+Account.m]: Pointer style

**Comment by JRHeaton** [comment]
:+1:

**Pull request merged**